tests/components/matrix/test_rooms.py

```diff
@@ -1,14 +1,36 @@
 """Test MatrixBot._join."""
 
+import pytest
+
 from homeassistant.components.matrix import MatrixBot
+from homeassistant.components.matrix.const import DOMAIN as MATRIX_DOMAIN
+from homeassistant.components.notify import DOMAIN as NOTIFY_DOMAIN
+from homeassistant.const import EVENT_HOMEASSISTANT_START
+from homeassistant.core import HomeAssistant
+from homeassistant.setup import async_setup_component
+
+from .conftest import MOCK_CONFIG_DATA
 
 from tests.components.matrix.conftest import TEST_BAD_ROOM, TEST_JOINABLE_ROOMS
 
 
-async def test_join(hass, matrix_bot: MatrixBot, caplog):
+async def test_join(
+    hass: HomeAssistant,
+    caplog: pytest.LogCaptureFixture,
+    mock_client,
+    mock_save_json,
+    mock_allowed_path,
+) -> None:
     """Test joining configured rooms."""
+    assert await async_setup_component(hass, MATRIX_DOMAIN, MOCK_CONFIG_DATA)
+    assert await async_setup_component(hass, NOTIFY_DOMAIN, MOCK_CONFIG_DATA)
```

Comment:
This needed to happen after `caplog` to be able to capture the log messages in time